sdv/sampling/independent_sampler.py

```diff
@@ -0,0 +1,172 @@
+            Dictionary mapping each table to its size. Should be instantiated and passed to the
+            sampler by the multi-table synthesizer using this sampler.
+    """
+
+    def __init__(self, metadata, table_synthesizers, table_sizes):
+        self.metadata = metadata
+        self._table_synthesizers = table_synthesizers
+        self._table_sizes = table_sizes
+
+    def _add_foreign_key_column(self, child_table, parent_table, child_name, parent_name):
+        """Add the foreign key that connects the child table to the parent table.
+
+        Args:
+            child_table (pd.DataFrame):
+                The table containing data sampled for the child.
+            parent_table (pd.DataFrame):
+                The table containing data sampled for the parent.
+            parent_name (str):
+                The name of the parent table.
+            child_name (str):
+                The name of the child table.
+        """
+        raise NotImplementedError()
+
+    def _get_all_foreign_keys(self, table_name):
+        foreign_keys = []
+        for relation in self.metadata.relationships:
+            if table_name == relation['child_table_name']:
+                foreign_keys.append(deepcopy(relation['child_foreign_key']))
+
+        return foreign_keys
+
+    def _process_samples(self, table_name, sampled_rows):
+        """Process the ``sampled_rows`` for the given ``table_name``.
+
+        Process the raw samples and convert them to the original space by reverse transforming
+        them. Also, when there are synthesizer columns (columns used to recreate an instance
+        of a synthesizer), those will be returned together.
+
+        Args:
+            table_name (str):
+                The name of the table.
+            sampled_rows (pd.DataFrame):
+                The sampled data to process.
+        """
+        data_processor = self._table_synthesizers[table_name]._data_processor
+        sampled = data_processor.reverse_transform(sampled_rows)
+
+        synthesizer_columns = list(set(sampled_rows.columns) - set(sampled.columns))
+        if synthesizer_columns:
+            sampled = pd.concat([sampled, sampled_rows[synthesizer_columns]], axis=1)
+
+        return sampled
+
+    def _sample_table(self, synthesizer, table_name, num_rows, sampled_data):
+        """Sample a single table and all its children.
+
+        Args:
+            synthesizer (SingleTableSynthesizer):
+                Synthesizer to sample from for the table.
+            table_name (string):
+                Name of the table to sample.
+            num_rows (int):
+                Number of rows to sample for the table.
+            sampled_data (dict):
+                A dictionary mapping table names to sampled tables (pd.DataFrame).
+        """
+        LOGGER.info('Sampling %s rows from table %s', num_rows, table_name)
+
+        if synthesizer._model:
+            sampled_rows = synthesizer._sample(num_rows)
```

Comment:
same potential problem about reject sampling here as in [other PR](https://github.com/sdv-dev/SDV/pull/1411#discussion_r1194410703)